aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -62,21 +63,16 @@ void cpu_index_kernel(TensorIterator& iter, IntArrayRef index_size, IntArrayRef
                       const func_t& f, bool serial_execution=false)
 {
   int ntensor = iter.ntensors();
+  const int index_parallel_grain_size = 3000;
```

Comment:
This is based on the DLRM experience and the test unit I written. As above one, it need a more time to analysis the threads launch overhead and computations relation to find a more perfect grain size, currently I didn't dig into that. Thanks

Reply:
More extensive benchmarking for different sizes will be valuable, if you've already done this, please let us know.

Reply:
Test result updated according to the lastest pytorch code

Reply:
So why would this PR improve single-threaded performance? Is it just flakyness of the benchmark? Also, why original number for 1 core and 28 cores different - they should be the same?

Reply:
This is most likely variance , for single-core, it will go into the previous sequence index_put and not to use atomic add float, as I add an if condition there:
 ```
bool use_parallel_for = ((iter.numel() >= internal::GRAIN_SIZE) && (at::get_num_threads() > 1));
if (iter.dtype() == at::ScalarType::Float && use_parallel_for) {
      //cpu_atomic_add_float path
} else {
     //original path
}
```

I enhance the test case with running 100 iterations instead of 10 to get the average performance on 100 iteration and updated as above. And why the modified code look a little better, it may due to code layout change